.github/workflows/publish_to_pypi.yml

```
@@ -0,0 +1,64 @@
+# Publish archives to PyPI and TestPyPI using GitHub Actions
+name: Publish to PyPI
+
+# Only run for tagged releases and pushes to the development branch
+on:
+  release:
+    types:
+      - published
+  push:
+    branches:
+      - development
+  # Runs for pull requests should be disabled other than for testing purposes
+  pull_request:
+    branches:
+      - development
```

Note to self: Comment these lines out before this PR is merged, as they are for testing purposes only (check that publishing to TestPyPI works.

```suggestion
  # pull_request:
  #   branches:
  #     - development
```

---

Is there a way to have this triggered if we put a certain keyword in a commit message to the development branch? I think that would be a clearer way to trigger test builds to PyPI than having to comment out these lines each time.

---

Yes, there are ways to trigger builds manually [using slash commands](https://github.com/marketplace/actions/slash-commands) (e.g. `/push-to-testpypi`) but it involves a bit more work.

I think it might be easier to have a 'Run workflow' button to press instead (see https://github.blog/changelog/2020-07-06-github-actions-manual-triggers-with-workflow_dispatch/). I'll try and set it up and test if it'll work to push to TestPyPI.

---

> I think it might be easier to have a 'Run workflow' button to press instead

Awesome! That sounds great to me.

Thanks for clarifying - I meant test builds to test-pypi but did not say that clearly.

---

Can we specify that it should only do a test pypi build when it's manually triggered? Otherwise, if the test build is successful it will automatically go on to do the PyPI build, even if it's not a tagged release. Or am I misunderstanding something?

---

Good question. The publishing to PyPI won't be done on manual triggers, it will only be done on published releases because of an `if` clause here:

https://github.com/icesat2py/icepyx/blob/1222a031f92bb4338428f9bdc47885a6efcceee2/.github/workflows/publish_to_pypi.yml#L58-L60

In other words, this publish to PyPI step is only run on tagged pushes.

---

I somehow managed to not register that. Thanks for pointing it out!